libs/CovidDatasets.py

```diff
@@ -67,14 +70,39 @@
     'Wyoming': 'WY'
 }
 
-class Dataset:
-    """Base class of the Dataset objects. Standardizes the output so that data from multiple differet sources can 
-    be fed into the model with as little hassle as possible."""
 
-    _POPULATION_URL = r'https://raw.githubusercontent.com/covid-projections/covid-data-public/master/data/misc/populations.csv'
-    _BEDS_URL = r'https://raw.githubusercontent.com/covid-projections/covid-data-public/master/data/beds-kff/beds.csv'
+local_public_data = tempfile.TemporaryDirectory()
+local_public_data_populated = False
+# public_data_base_url = "https://raw.githubusercontent.com/covid-projections/covid-data-public/master"
+public_data_base_url = 'file://localhost' + local_public_data.name + '/covid-data-public-master'
+
+
+def create_local_copy_public_data():
+    """
+    Creates a local copy of the public data repository. This is done to avoid
+    downloading the file again for each intervention type.
+    """
+    github_zip_url = "https://github.com/covid-projections/covid-data-public/archive/master.zip"
+    logging.info("Creating a Local Copy of {}".format(github_zip_url))
+
+    zipresp = urlopen(github_zip_url)
+    with open(local_public_data.name + "/master.zip", "wb") as tempzip:
+        tempzip.write(zipresp.read())
+
+    with ZipFile(local_public_data.name + "/master.zip") as zf:
+        zf.extractall(path=local_public_data.name)
+
+    local_public_data_populated = True
 
 
+class Dataset:
+    """Base class of the Dataset objects. Standardizes the output so that data
+    from multiple differenmt sources can be fed into the model with as little
+    hassle as possible."""
+
+    _POPULATION_URL = public_data_base_url + '/data/misc/populations.csv'
```

Comment:
It would be great if we could pass this in as an argument to the init function (since it can change).

I've sometimes made these default arugments:

```
class Dataset:
    def __init__(self, start_date, filter_past_date=None, population_url=DEFAULT_URL):
        self.population_url = population_url
        ...

```

Reply:
I had thought about this, but didn't want to make the call. The other thought I had was to use an env_var to allow switching. It is pretty easy to shift this around.